### PR TITLE
Fix resizeMode="cover" when image is resized

### PR DESCRIPTION
**Pull request by christophpurrer**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

Since `RCTResizeModeCover` is implemented manually on macOS (iOS benefits from `UIViewContentModeScaleAspectFill` API), we need to reprocess the image any time the bounds of the view update. The `updateImage:` method was already called on resize, so this instead stores the original image in an ivar so it can call `RCTFillImagePreservingAspectRatio` from `updateImage:` instead.

## Changelog

[macOS] [Fixed] - Fix resizeMode="cover" when image is resized

### Before

Look at the images labeled 'Cover' and observe how they don't properly resize

https://user-images.githubusercontent.com/484044/180457216-88a05f6f-3aef-40a6-af45-8e3e2f4feef7.mov

### After

https://user-images.githubusercontent.com/484044/180457246-2b09e140-1dfe-489c-8d43-56bff5e53740.mov

### Before
Another demo, done internally

https://user-images.githubusercontent.com/484044/180457315-ea46deba-a9a1-4719-bbc1-7e495a19792e.mov

### After

https://user-images.githubusercontent.com/484044/180457545-78eb1ab3-2401-48e6-b038-a225fee71c33.mov

### CODE REVIEW

File: Libraries/Image/RCTImageView.mm [hunk 1]
```diff
@@ -233,17 +241,13 @@ - (void)setImage:(UIImage *)image
...
-    }
-#endif // ]TODO(macOS GH#774)
+    _image = image; // TODO(macOS GH#774)
     [self updateWithImage:image];
   }
 }
 
 - (UIImage *)image {
-  return _imageView.image;
+  return _image ?: _imageView.image; // TODO(macOS GH#774)
```
Comment (harrieshin) [inline Libraries/Image/RCTImageView.mm:250]: what would be the case that imag_view.image will be set when _image is not?

**Comment by christophpurrer** [comment]
@mischreiber > Does this change look good to you?

**Comment by Saadnajmi** [comment]
"Needs attention" label here means that we (Microsoft, aka @mischreiber) will take a deeper look. No changes requested from the PR author yet.

**Comment by christophpurrer** [comment]
Anything needed from our side to get this change landed?

**Comment by Saadnajmi** [comment]
> Anything needed from our side to get this change landed?

@mischrieber had wanted to take a deeper look, but is also currently OOF. Is this is blocking other changes from making their way through, we can prioritize this one over others. Come to think of it, it probably takes priority by default on account of being "older" than the others..

File: Libraries/Image/RCTImageView.mm [hunk 2]
```diff
@@ -128,6 +128,8 @@ @implementation RCTImageView
   // Whether the latest change of props requires the image to be reloaded
   BOOL _needsReload;
 
+  UIImage *_image; // TODO(macOS GH#774)
```
Comment (Saadnajmi) [inline Libraries/Image/RCTImageView.mm:131]: Dumb question, why isn't this an NSImage?

Comment (Saadnajmi) [reply Libraries/Image/RCTImageView.mm:131]: Seems to be the case in the rest of this file.

**Review by Saadnajmi — APPROVED** [review]
I'm going to go ahead and also approve this. It's an old enough , simple PR that I'd rather not linger. We can back it out if it causes issues, @mischreiber FYI

**Pull request merged**